src/MapImager.cpp

```diff
@@ -97,19 +97,19 @@ void MapImager::LoadTilesets(MapData& mapData, RenderManager& mapImager, bool ac
 
 		string tilesetFilename(mapData.tilesetSources[i].tilesetFilename + ".bmp");
 
-		//TODO: Allow FreeImage to take a pointer to the associated well within a vol file 
-		//      instead of forcing its extraction.
-		bool extracted = resourceManager.ExtractSpecificFile(tilesetFilename);
-		
-		if (!extracted) {
+		auto stream = resourceManager.GetResourceStream(tilesetFilename);
+
+		if (stream == nullptr) {
 			throw runtime_error("Unable to find the tileset " + tilesetFilename + " in the directory or in a given archive (.vol).");
 		}
 
-		mapImager.AddTileset(tilesetFilename, ImageFormat::BMP);
+		if (stream->Length() > UINT32_MAX) {
+			throw std::runtime_error("Tileset " + tilesetFilename + " is too large for OP2MapImager to load into memory");
+		}
```

Comment:
I updated it to use SIZE_MAX from cstdint, which I believe is the same result as std::numeric_limits<type>::max(). Not sure that one is better over the other?

Reply:
Yes, it's the same numeric value. The difference is in the code maintenance.

```cpp
assert(SIZE_MAX == std::numeric_limits<size_t>::max());
assert(UINT_MAX == std::numeric_limits<unsigned int>::max());
```

Note how the constant name must correspond with the type name, but there is no guarantees they will match up well. This is particularly true with code refactoring, or if a data type is changed. Contrast that with:

```cpp
auto sourceSize = ...;
type targetSize;

if (sourceSize > std::numeric_limits<targetSize>::max()) {
  throw std::runtime_error("Source value out of range for target");
}
targetSize = static_cast<decltype(targetSize)>(sourceSize);
```

In the above, you can freely change `type` and it will still be correct.